From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: A Mak <refragable@mailbox.org>
Date: Sat, 8 Aug 2020 11:17:59 -0700
Subject: [PATCH] disable learn more link in incognito new tab

Change since M139:

Now also hides the not applicable third party cookie card text.
---
 .../chrome/browser/ntp/IncognitoDescriptionView.java       | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoDescriptionView.java b/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoDescriptionView.java
index 5abfcc0d50406..dedd9b0795dd6 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoDescriptionView.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoDescriptionView.java
@@ -371,8 +371,8 @@ public class IncognitoDescriptionView extends LinearLayout {
         final ChromeClickableSpan learnMoreSpan =
                 new ChromeClickableSpan(mLearnMore.getSpanColor(), (_) -> mLearnMore.callOnClick());
 
-        boolean learnMoreInSubtitle = mWidthDp > WIDE_LAYOUT_THRESHOLD_DP;
-        mLearnMore.setVisibility(learnMoreInSubtitle ? View.GONE : View.VISIBLE);
+        boolean learnMoreInSubtitle = false;
+        mLearnMore.setVisibility(View.GONE);
 
         if (!learnMoreInSubtitle) {
             // Format the "Learn more" link.
@@ -408,6 +408,9 @@ public class IncognitoDescriptionView extends LinearLayout {
         @Nullable ViewGroup cookieControlsCard = findViewById(R.id.tracking_protection_card);
         // Still null - not inflated yet.
         if (cookieControlsCard == null) return;
+        {
+            cookieControlsCard.setVisibility(View.GONE);
+        }
         if (mWidthDp <= WIDE_LAYOUT_THRESHOLD_DP) {
             // Portrait
             cookieControlsCard.getLayoutParams().width = LinearLayout.LayoutParams.MATCH_PARENT;
